library/components/TNavigationDropdown.vue

```diff
@@ -100,11 +100,36 @@ export default {
     selected() {
       return window.location.pathname.substring(1);
     },
+    pages() {
+      const parentContextUrl = this.whoami?.context?.parent;
+      const pages = [];
+      for (let page of this.project.nav) {
+        page.main_url = page.url; // Common parameter for 'active page' check.
```

Comment:
Why do we need to create this `main_url` attribute if it's equal to `page.url`?

Reply:
main_url actually just stores the main path of the page (issues-summary or issues-detail) while the url can get modified if inside iframe, see line 110.

Reply:
Basically the logic is, if inside iframe then modify the default url to also use paren't full url but if anchor tag is just clicked instead of using context menu to open as new tab then `redirectToPage` function is used.